### PR TITLE
KOGITO-2214: Integrate Task Form Building Block with the runtime endpoints

### PR DESCRIPTION
**Pull request by pefernan**
Adapting the current TaskForm component to use the new rutnimes forms endpoints. It also adapts the the form schema to the new format.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by pefernan** [comment]
@jstastny-cz @nmirasch could you please take a look?

File: ui-packages/packages/task-console/src/components/Atoms/FormNotification/tests/__snapshots__/FormNotification.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+                            <button
+                              aria-disabled={null}
+                              aria-label="Close"
+                              className="pf-c-button pf-m-plain"
+                              disabled={false}
+                              onClick={[MockFunction]}
+                              tabIndex={null}
+                              type="button"
+                            >
+                              <TimesIcon
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Atoms/FormNotification/tests/__snapshots__/FormNotification.test.tsx.snap:304]: we could mock the icon

File: ui-packages/packages/task-console/src/components/Atoms/FormFooter/FormFooter.tsx [hunk 2]
```diff
@@ -12,17 +11,20 @@ interface IOwnProps {
...
       {actions.map(action => {
         return (
           <Button
             key={'submit-' + action.name}
             type="submit"
-            variant={action.primary ? 'primary' : 'secondary'}
             onClick={action.onActionClick}
           >
-            {action.name}
+            {capitalize(action.name)}
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/components/Atoms/FormFooter/FormFooter.tsx:28]: should this still be called action? I've seen lots of changes from action to phase, is this something different?

Comment (pefernan) [reply ui-packages/packages/task-console/src/components/Atoms/FormFooter/FormFooter.tsx:28]: In this case it makes sense, I'd like the footer only being able to execute the actions provided the renderer so its not aware of the runtime specifics.

**Comment by pefernan** [comment]
@jstastny-cz finally I moved the submit logic into a separate class so the renderer isn't aware of the runtime internals.

**Comment by pefernan** [comment]
jenkins retest this please

File: ui-packages/packages/task-console/src/util/uniforms/TaskFormSubmitHandler/TaskFormSubmitHandler.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    this.selectedPhase = phase;
+  }
+
+  getActions = () => {
+    return this.actions;
+  }
+
+  doSubmit = async (formData: any) => {
+
+    if(this.actions.length === 0 || !this.selectedPhase) {
```
Comment (jstastny-cz) [inline ui-packages/packages/task-console/src/util/uniforms/TaskFormSubmitHandler/TaskFormSubmitHandler.ts:74]: whitespace inconsistency - once `if(` below is `if (`

**Review by jstastny-cz — APPROVED** [review]
just a small formatting issue, otherwise looks good.

File: ui-packages/packages/task-console/src/util/uniforms/TaskFormSubmitHandler/TaskFormSubmitHandler.ts (hunk 3 above)
Comment (pefernan) [reply ui-packages/packages/task-console/src/util/uniforms/TaskFormSubmitHandler/TaskFormSubmitHandler.ts:74]: @jstastny-cz cool. I'll wait for @nmirasch's review to do the change along with her suggestions.

File: ui-packages/packages/task-console/src/util/uniforms/TaskFormSubmitHandler/tests/TaskFormSubmitHandler.test.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+ * limitations under the License.
+ */
+import _ from 'lodash';
+import axios from 'axios';
+
+import ApplyForVisaForm from '../../../tests/mocks/ApplyForVisa';
+import { UserTaskInstance } from '../../../../graphql/types';
+import { TaskInfoImpl } from '../../../../model/TaskInfo';
+import { TaskFormSubmitHandler } from '../TaskFormSubmitHandler';
+/*import { mount } from 'enzyme';
```
Comment (nmirasch) [inline ui-packages/packages/task-console/src/util/uniforms/TaskFormSubmitHandler/tests/TaskFormSubmitHandler.test.ts]: @pefernan  This comment could be removed?

**Comment by nmirasch** [comment]
@pefernan  just a minor comment regarding commented code, but otherwise looks good to me!

**Review by nmirasch — APPROVED** [review]

**Comment by pefernan** [comment]
Jenkins retest this please

**Comment by pefernan** [comment]
Jenkins retest this please

**Pull request merged**